### PR TITLE
Add changelog to documentation

### PR DESCRIPTION
**Pull request by tmichela**
Adds a changelog page to docs since we now version and publish the package.

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+- Plot xarray (!289).
+
+Changed:
+
+- Preserve context file path for tracebacks (!249).
+- Improve error handling when opening variables in the GUI (!257).
+- Evaluate non-cluster variables in Slurm using Solaris cluster (!256).
+- Use Slurm jobs for reprocessing (!270).
+- HDF5 array compression (!273).
+- Simplify testing CLI by passing argv in (!279).
```
Comment (takluyver) [inline docs/changelog.md:51]: I generally think changelogs should describe user-visible changes, so I tend to leave out things about testing, CI, and most docs changes. YMMV :slightly_smiling_face:

**Comment by takluyver** [comment]
Are we aiming to record application changes going forwards? Or only changes affecting the API, on the grounds that that's why we started doing numbered versions, and we routinely deploy the application between 'releases'?

File: docs/changelog.md (hunk 1 above)
Comment (tmichela) [reply docs/changelog.md]: right, I did remove quite a bit already, some may have slip through. I cleaned a bit more now

**Comment by tmichela** [comment]
I would be in favor to recording application changes, as people primarily go to the documentation for the GUI usage, how to write a context file and how to run the backend.

**Comment by takluyver** [comment]
OK, that makes sense. Is it worth also putting the date each version was released, so you can get a rough sense of when changes were made? E.g. I [do this for EXtra-data](https://extra-data.readthedocs.io/en/latest/changelog.html).

Other than that, LGTM

**Comment by tmichela** [comment]
Thanks, added the release dates.

**Pull request merged**